### PR TITLE
fix(codegen): typed-array dispatch holes in push/<</reverse!/sort!

### PR DESCRIPTION
**Pull request by grepsedawk**
## Summary

- `push`, `<<`, `reverse!`, and `sort!` silently fell through for non-IntArray typed-array variants — `floats.reverse!`, `strs.sort!`, `syms << x` and friends were no-ops.
- Extends each handler to dispatch for IntArray/SymArray/StrArray/FloatArray (plus PtrArray for `push`/`<<`), and adds the missing runtime helpers (`sp_FloatArray_reverse_bang`, `sp_StrArray_sort_bang`, etc.).
- Three regression tests under `test/`; two of them fail on `master` without this fix.

## What was tried and skipped

A dispatcher that uses `array_c_prefix` to fold the per-type branches into a single helper call was attempted, but broke bootstrap — an unrelated compound boolean expression at `compile_call_expr:14750` came out as malformed C in the gen2 output. Reverted.

A template-driven version of the runtime helpers was also considered. The per-type variation (IntArray's deque-style `start` offset, StrArray's GC scan callback, divergent `min`/`max`-on-empty behavior) is larger than it looks at first glance, and given the dispatcher attempt's regression, didn't want to push the self-host fragility further. Both are reasonable follow-ups in a separate PR.

## Note on `_sp_float_cmp` and NaN

`(va > vb) - (va < vb)` returns 0 for any NaN-involved comparison, so `qsort` ordering of arrays containing NaN is unspecified. This matches the existing `_sp_int_cmp` precedent (no NaN edge case) and CRuby's `Array#sort` doesn't guarantee anything sane with NaN either.

## Test plan

- [x] `make bootstrap` converges (gen2.c == gen3.c)
- [x] `make test`: 88 pass, 2 pre-existing fail (`bm_constant_path_builtin_new`, `bm_fiber`), 1 pre-existing error (`bm_sym_case`) — same as `master` plus 3 new passing tests
- [x] `bm_array_reverse_bang` and `bm_array_sort_bang` confirmed to fail on `master` without the fix
- [x] `bm_array_push` covers `push` + `<<` for all four typed-array variants

### CODE REVIEW

File: spinel_codegen.rb [hunk 1]
```diff
@@ -17505,7 +17505,7 @@ def compile_mutating_call_stmt(nid, mname, recv)
     if mname == "<<"
       if recv >= 0
         rt = infer_type(recv)
-        if rt == "int_array"
+        if rt == "int_array" || rt == "sym_array"
           rc = compile_expr_gc_rooted(recv)
           emit("  sp_IntArray_push(" + rc + ", " + compile_arg0(nid) + ");")
           return 1
```
[the remark this answers is not in the document]
Comment (grepsedawk) [reply spinel_codegen.rb:17525]: False positive — the `<<` handler does dispatch for str_array, float_array, and ptr_array (lines 17513-17527). The lambda-to-int cast inconsistency was real and is now fixed.

File: lib/sp_runtime.h [hunk 2]
```diff
@@ -208,6 +208,9 @@ static inline mrb_int sp_FloatArray_length(sp_FloatArray*a){return a->len;}
 static inline mrb_bool sp_FloatArray_empty(sp_FloatArray*a){return a->len==0;}
 static inline mrb_float sp_FloatArray_get(sp_FloatArray*a,mrb_int i){if(i<0)i+=a->len;return a->data[i];}
 static inline void sp_FloatArray_set(sp_FloatArray*a,mrb_int i,mrb_float v){if(i<0)i+=a->len;while(i>=a->cap){a->cap=a->cap*2+1;a->data=(mrb_float*)realloc(a->data,sizeof(mrb_float)*a->cap);}while(i>=a->len){a->data[a->len]=0.0;a->len++;}a->data[i]=v;}
+static void sp_FloatArray_reverse_bang(sp_FloatArray*a){for(mrb_int i=0,j=a->len-1;i<j;i++,j--){mrb_float t=a->data[i];a->data[i]=a->data[j];a->data[j]=t;}}
+static int _sp_float_cmp(const void*a,const void*b){mrb_float va=*(const mrb_float*)a,vb=*(const mrb_float*)b;return(va>vb)-(va<vb);}
+static void sp_FloatArray_sort_bang(sp_FloatArray*a){qsort(a->data,a->len,sizeof(mrb_float),_sp_float_cmp);}
```
[the remark this answers is not in the document]
Comment (grepsedawk) [reply lib/sp_runtime.h:213]: Fixed — added `sp_PtrArray_reverse_bang` helper.

File: spinel_codegen.rb [hunk 3]
```diff
@@ -17680,33 +17680,66 @@ def compile_mutating_call_stmt(nid, mname, recv)
...
+        if rt == "str_array"
+          rc = compile_expr_gc_rooted(recv)
+          emit("  sp_StrArray_reverse_bang(" + rc + ");")
+          return 1
+        end
+        if rt == "float_array"
+          rc = compile_expr_gc_rooted(recv)
+          emit("  sp_FloatArray_reverse_bang(" + rc + ");")
+          return 1
+        end
```
[the remark this answers is not in the document]
Comment (grepsedawk) [reply spinel_codegen.rb:17735]: Fixed — added ptr_array dispatch (refactored as a prefix-based form to dodge an unrelated self-host miscompile in this method).
